### PR TITLE
Services implement HealthReport 

### PR DESCRIPTION
**Pull request by essamhassan**
This PR implements HealthReport for all services in core node.
Current implementation for HealthReport is a drop in replacement for Healthy()

This code is not used yet, it doesn't make sense to add unit tests in the current state since all the logic within HealthReport is going to be modified to incl more granular checks.

Next steps:
- Go over srvcs
- Define Healthy() correctly per service
- Implement HealthReport to return health status for the service as well
  as it's subservices.

### CODE REVIEW

**Review by samsondav — APPROVED** [review]

File: core/services/pipeline/runner.go [hunk 1]
```diff
@@ -142,6 +142,14 @@ func (r *runner) Close() error {
 	})
 }
 
+func (r *runner) Name() string {
+	return r.lggr.Name()
+}
+
+func (r *runner) HealthReport() map[string]error {
+	return map[string]error{}
```
Comment (jmank88) [inline core/services/pipeline/runner.go:150]: Should this one have the usual default?

File: core/services/promreporter/prom_reporter.go [hunk 2]
```diff
@@ -125,6 +125,13 @@ func (pr *promReporter) Close() error {
 		return nil
 	})
 }
+func (pr *promReporter) Name() string {
+	return pr.lggr.Name()
+}
+
+func (pr *promReporter) HealthReport() map[string]error {
+	return map[string]error{}
```
Comment (samsondav) [inline core/services/promreporter/prom_reporter.go:133]: Same here

File: core/services/job/spawner.go [hunk 3]
```diff
@@ -110,6 +110,14 @@ func (js *spawner) Close() error {
 	})
 }
 
+func (js *spawner) Name() string {
+	return js.lggr.Name()
+}
+
+func (js *spawner) HealthReport() map[string]error {
+	return map[string]error{}
```
Comment (jmank88) [inline core/services/job/spawner.go:118]: Default?

File: core/services/ocrcommon/peer_wrapper.go [hunk 4]
```diff
@@ -217,6 +217,14 @@ func (p *SingletonPeerWrapper) Close() error {
 	})
 }
 
+func (p *SingletonPeerWrapper) Name() string {
+	return p.lggr.Name()
+}
+
+func (p *SingletonPeerWrapper) HealthReport() map[string]error {
+	return map[string]error{}
```
Comment (jmank88) [inline core/services/ocrcommon/peer_wrapper.go:225]: Default?

Comment (essamhassan) [reply core/services/ocrcommon/peer_wrapper.go]: Fixed.

File: core/services/promreporter/prom_reporter.go (hunk 2 above)
Comment (essamhassan) [reply core/services/promreporter/prom_reporter.go]: Fixed

File: core/services/pipeline/runner.go (hunk 1 above)
Comment (essamhassan) [reply core/services/pipeline/runner.go]: Fixed

**Review by jmank88 — APPROVED** [review]

**Pull request merged**